apps/www/registry/default/ui/timeline.tsx

```diff
@@ -0,0 +1,205 @@
+
+interface TimelineProps
+  extends React.HTMLAttributes<HTMLUListElement>,
+    VariantProps<typeof timelineVariants> {}
+
+const Timeline = React.forwardRef<HTMLUListElement, TimelineProps>(
+  ({ children, className, positions, ...props }, ref) => {
+    return (
+      <ul
+        className={cn(timelineVariants({ positions }), className)}
+        ref={ref}
+        {...props}
+      >
+        {children}
+      </ul>
+    )
+  }
+)
+Timeline.displayName = "Timeline"
+
+const timelineItemVariants = cva("grid items-center gap-x-2", {
+  variants: {
+    status: {
+      done: "text-primary",
+      default: "text-muted-foreground",
+    },
+  },
+  defaultVariants: {
+    status: "default",
+  },
+})
+
+interface TimelineItemProps
+  extends React.HTMLAttributes<HTMLLIElement>,
+    VariantProps<typeof timelineItemVariants> {}
+
+const TimelineItem = React.forwardRef<HTMLLIElement, TimelineItemProps>(
+  ({ className, status, ...props }, ref) => (
+    <li
+      className={cn(timelineItemVariants({ status }), className)}
+      ref={ref}
+      {...props}
+    />
+  )
+)
+TimelineItem.displayName = "TimelineItem"
+
+const timelineDotVariants = cva(
+  "col-start-2 col-end-3 row-start-1 row-end-1 flex size-4 items-center justify-center rounded-full border border-current",
+  {
+    variants: {
+      status: {
+        default: "[&>svg]:hidden",
+        current:
+          "[&>.lucide-circle]:fill-current [&>.lucide-circle]:text-current [&>svg:not(.lucide-circle)]:hidden",
+        done: "bg-primary [&>.lucide-check]:text-background [&>svg:not(.lucide-check)]:hidden",
+        error:
+          "border-destructive bg-destructive [&>.lucide-x]:text-background [&>svg:not(.lucide-x)]:hidden",
+      },
+    },
+    defaultVariants: {
+      status: "default",
+    },
+  }
+)
+
+interface TimelineDotProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    VariantProps<typeof timelineDotVariants> {
+  customIcon?: React.ReactNode
+}
+
+const TimelineDot = React.forwardRef<HTMLDivElement, TimelineDotProps>(
+  ({ className, status, customIcon, ...props }, ref) => (
+    <div
+      role="status"
+      className={cn("timeline-dot", timelineDotVariants({ status }), className)}
+      ref={ref}
+      {...props}
+    >
+      <Circle className="size-2.5" />
+      <Check className="size-3" />
+      <X className="size-3" />
+      {customIcon}
+    </div>
+  )
+)
+TimelineDot.displayName = "TimelineDot"
+
+const timelineContentVariants = cva(
+  "row-start-2 row-end-2 pb-8 text-muted-foreground",
+  {
+    variants: {
+      side: {
+        right: "col-start-3 col-end-4 mr-auto text-left",
+        left: "col-start-1 col-end-2 ml-auto text-right",
+      },
+    },
+    defaultVariants: {
+      side: "right",
+    },
+  }
+)
+
+interface TimelineConent
+  extends React.HTMLAttributes<HTMLParagraphElement>,
+    VariantProps<typeof timelineContentVariants> {}
+
+const TimelineContent = React.forwardRef<HTMLParagraphElement, TimelineConent>(
```

Comment:
I believe this one is a typo

```suggestion
interface TimelineContent
  extends React.HTMLAttributes<HTMLParagraphElement>,
    VariantProps<typeof timelineContentVariants> {}

const TimelineContent = React.forwardRef<HTMLParagraphElement, TimelineContent>(
```